pkg/solana/chainwriter/chain_writer.go

```diff
@@ -0,0 +1,324 @@
+			if err != nil {
+				continue
+			}
+
+			// Collect public keys that are actually used
+			var usedAddresses solana.PublicKeySlice
+			for _, meta := range metas {
+				if _, exists := usedAccounts[meta.PublicKey.String()]; exists {
+					usedAddresses = append(usedAddresses, meta.PublicKey)
+				}
+			}
+
+			// Add to the filtered map if there are any used addresses
+			if len(usedAddresses) > 0 {
+				filteredLookupTables[tableKey] = usedAddresses
+			}
+		}
+	}
+
+	// Filter static lookup tables
+	for tableKey, addresses := range staticTableMap {
+		var usedAddresses solana.PublicKeySlice
+		for _, staticAddress := range addresses {
+			if _, exists := usedAccounts[staticAddress.String()]; exists {
+				usedAddresses = append(usedAddresses, staticAddress)
+			}
+		}
+
+		// Add to the filtered map if there are any used addresses
+		if len(usedAddresses) > 0 {
+			filteredLookupTables[tableKey] = usedAddresses
+		}
+	}
+
+	return filteredLookupTables
+}
+
+func (s *SolanaChainWriterService) SubmitTransaction(ctx context.Context, contractName, method string, args any, transactionID string, toAddress string, meta *types.TxMeta, value *big.Int) error {
+	programConfig, exists := s.config.Programs[contractName]
+	if !exists {
+		return fmt.Errorf("failed to find program config for contract name: %s", contractName)
+	}
+	methodConfig, exists := programConfig.Methods[method]
+	if !exists {
+		return fmt.Errorf("failed to find method config for method: %s", method)
+	}
+
+	// Configure debug ID
+	debugID := ""
+	if methodConfig.DebugIDLocation != "" {
+		var err error
+		debugID, err = GetDebugIDAtLocation(args, methodConfig.DebugIDLocation)
+		if err != nil {
+			return errorWithDebugID(fmt.Errorf("error getting debug ID from input args: %w", err), debugID)
+		}
+	}
+
+	codec := s.codecs[contractName]
+	encodedPayload, err := codec.Encode(ctx, args, method)
+	if err != nil {
+		return errorWithDebugID(fmt.Errorf("error encoding transaction payload: %w", err), debugID)
+	}
+
+	// Fetch derived and static table maps
+	derivedTableMap, staticTableMap, err := s.ResolveLookupTables(ctx, args, methodConfig.LookupTables)
+	if err != nil {
+		return errorWithDebugID(fmt.Errorf("error getting lookup tables: %w", err), debugID)
+	}
+
+	// Resolve account metas
+	accounts, err := GetAddresses(ctx, args, methodConfig.Accounts, derivedTableMap, s.reader)
+	if err != nil {
+		return errorWithDebugID(fmt.Errorf("error resolving account addresses: %w", err), debugID)
+	}
+
+	// Filter the lookup table addresses based on which accounts are actually used
+	filteredLookupTableMap := s.FilterLookupTableAddresses(accounts, derivedTableMap, staticTableMap)
+
+	// Fetch latest blockhash
+	blockhash, err := s.reader.LatestBlockhash(ctx)
+	if err != nil {
+		return errorWithDebugID(fmt.Errorf("error fetching latest blockhash: %w", err), debugID)
+	}
+
+	// Prepare transaction
+	programID, err := solana.PublicKeyFromBase58(contractName)
```

Comment:
I think this should be `toAddress` based on our conversation in the other thread.
```suggestion
	programID, err := solana.PublicKeyFromBase58(toAddress)
```

Reply:
Yep, good catch